perfkitbenchmarker/linux_benchmarks/hpcg_benchmark.py

```diff
@@ -0,0 +1,292 @@
+# Copyright 2017 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Run HPCG.
+
+Requires:
+* openmpi 1.10.2 or openmpi 1.6.5
+
+Binaries are provided which are statically linked to CUDA 8.0.44 and
+OpenMPI 1.10.2 and 1.6.5. If different versions of these libs are to be used,
+modify the HPCG package to build from source, located here:
+https://github.com/hpcg-benchmark/hpcg/
+"""
+
+import logging
+import os
+import re
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import data
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import flag_util
+from perfkitbenchmarker import sample
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.linux_packages import cuda_toolkit_8
+from perfkitbenchmarker.linux_packages import hpcg
+
+FLAGS = flags.FLAGS
+RUN_SCRIPT = 'run_hpcg'
+CONFIG_FILE = 'hpcg.dat'
+MACHINEFILE = 'HOSTFILE'
+
+BENCHMARK_VERSION = 0.1
+BENCHMARK_NAME = 'hpcg'
+BENCHMARK_CONFIG = """
+hpcg:
+  description: Runs HPCG. Specify the number of VMs with --num_vms
+  flags:
+    gce_migrate_on_maintenance: False
+    image_project: ubuntu-os-cloud
+  vm_groups:
+    default:
+      vm_spec:
+        GCP:
+          image: ubuntu-1604-xenial-v20170307
+          machine_type: n1-standard-4-k80x1
+          zone: us-east1-d
+          boot_disk_size: 200
+        AWS:
+          image: ami-a9d276c9
+          machine_type: p2.xlarge
+          zone: us-west-2b
+          boot_disk_size: 200
+        Azure:
+          image: Canonical:UbuntuServer:16.04.0-LTS:latest
+          machine_type: Standard_NC6
+          zone: eastus
+      vm_count: null
+"""
+
+flags.DEFINE_integer(
+    'hpcg_runtime', 60, 'hpcg runtime in seconds', lower_bound=1)
+
+flags.DEFINE_integer(
+    'hpcg_gpus_per_node', None, 'The number of gpus per node.', lower_bound=1)
+
+flag_util.DEFINE_integerlist(
```

Comment:
This is used to set the number of mpi processes in `run_hpcg.sh`. `hpcg.dat` has no effect on the number of gpus used.